src/poly1305.rs

```diff
@@ -87,27 +191,62 @@ pub type Tag = [u8; TAG_LEN];
 /// The length of a `Tag`.
 pub const TAG_LEN: usize = 128 / 8;
 
-type State = [u8; STATE_LEN];
-const STATE_LEN: usize = 256;
+const BLOCK_STATE_SIZE: usize = 192;
```

Comment:
Let's use `_LEN` for lengths.

Question: Why do you prefix these names with `BLOCK_`? Why not `STATE_LEN` and `State`?